CIP2/CIP2.md

```diff
@@ -0,0 +1,983 @@
+---
+CIP: 2
+Title: Coin Selection Algorithms for Cardano
+Author: Jonathan Knowles <jonathan.knowles@iohk.io>
```

Comment:
I'd also mention Edsko De Vries in the authors here since he's the one who mostly came up with the original design.

Reply:
> I'd also mention Edsko De Vries in the authors here since he's the one who mostly came up with the original design.

Thanks for this suggestion!

I will ask @edsko if he wishes to be attributed here.

Reply:
@KtorZ wrote:
> I'd also mention Edsko De Vries in the authors here since he's the one who mostly came up with the original design.

I had an opportunity to ask @edsko about this offline.

He asked me to clarify the citation of his original article, but stated that it's not necessary to add him as an author to the specification, since although the specification does reference his work, it doesn't include content authored by him directly. I think he's happy so long as the citations clearly point out the authorship of the original algorithm. (I'm sure @edsko will correct me if I've misinterpreted.)

In response to @edsko's feedback, I've clarified the citation (see [Self Organisation in Coin Selection](https://github.com/jonathanknowles/CIPs/blob/CIP-JonathanKnowles-CoinSelectionAlgorithms/CIP2/CIP2.md#self-organisation-in-coin-selection)) to clearly point out the authorship of the Random-Improve algorithm.

Hopefully the abstract should be clear enough, but would be grateful to hear any feedback you may have, especially if you feel it could be even clearer.